exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/flatten/FlattenRecordBatch.java

```diff
@@ -129,11 +129,21 @@ private void setFlattenVector() {
     try {
       final TypedFieldId typedFieldId = incoming.getValueVectorId(popConfig.getColumn());
       final MaterializedField field = incoming.getSchema().getColumn(typedFieldId.getFieldIds()[0]);
-      final RepeatedValueVector vector = RepeatedValueVector.class.cast(incoming.getValueAccessorById(
-          field.getValueClass(), typedFieldId.getFieldIds()).getValueVector());
+      RepeatedValueVector vector = null;
+      try {
+        vector = RepeatedValueVector.class.cast(incoming.getValueAccessorById(
+            field.getValueClass(), typedFieldId.getFieldIds()).getValueVector());
+      } catch(ClassCastException ex) {
```

Comment:
In stead of catch ClassCastException, you may check if the input value vector is an instance of RepeatedValueVector.  Using try{} catch{} seems a bit over-kill for this case.